src/mock/mock_mixin.ts

```diff
@@ -0,0 +1,164 @@
+import type { Constructor, MethodCalls, MethodOf } from "../types.ts";
+import { PreProgrammedMethod } from "../pre_programmed_method.ts";
+import type { IMock } from "../interfaces.ts";
+
+class MockError extends Error {}
+
+class MethodExpectation<OriginalObject> {
+  #method_name: MethodOf<OriginalObject>;
+  #expected_calls = 0;
+
+  get method_name(): MethodOf<OriginalObject> {
+    return this.#method_name;
+  }
+
+  get expected_calls(): number {
+    return this.#expected_calls;
+  }
+
+  constructor(methodName: MethodOf<OriginalObject>) {
+    this.#method_name = methodName;
+  }
+
+  public toBeCalled(expectedCalls: number) {
+    this.#expected_calls = expectedCalls;
+  }
+}
+
+export function createMock<OriginalConstructor, OriginalObject>(
+  OriginalClass: OriginalConstructor,
+): IMock<OriginalObject> {
+  const Original = OriginalClass as unknown as Constructor<
+    // deno-lint-ignore no-explicit-any
+    (...args: any[]) => any
+  >;
+  return new class MockExtension extends Original {
+    /**
+     * Helper property to see that this is a mock object and not the original.
+     */
+    is_mock = true;
+
+    /**
+     * Property to track method calls.
+     */
+    #calls!: MethodCalls<OriginalObject>;
+
+    /**
+     * An array of expectations to verify (if any).
+     */
+    #expectations: MethodExpectation<OriginalObject>[] = [];
+
+    /**
+     * The original object that this class creates a mock of.
+     */
+    #original!: OriginalObject;
+
+    //////////////////////////////////////////////////////////////////////////////
+    // FILE MARKER - CONSTRUCTOR /////////////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * @param original - The original object to mock.
+     * @param methodsToTrack - The original object's method to make trackable.
+     */
+    public init(original: OriginalObject, methodsToTrack: string[]) {
+      this.#original = original;
+      this.#calls = this.#constructCallsProperty(methodsToTrack);
+    }
```

Comment:
constructor needs to be updated with methods - public

Reply:
actually that is below so maybe rm that comment block and move method?

Reply:
moved them stuffs: [08e47ef](https://github.com/drashland/rhum/pull/150/commits/08e47efb29339ffe1b1a8044ef45b9e9b07bb361)